src/Features/LanguageServer/Protocol/Handler/Diagnostics/AbstractWorkspacePullDiagnosticsHandler.cs

```diff
@@ -37,14 +36,16 @@ internal abstract class AbstractWorkspacePullDiagnosticsHandler<TDiagnosticsPara
         IDiagnosticsRefresher diagnosticRefresher,
         IGlobalOptionService globalOptions) : base(diagnosticAnalyzerService, diagnosticRefresher, globalOptions)
     {
-        _diagnosticSourceManager = diagnosticSourceManager;
+        DiagnosticSourceManager = diagnosticSourceManager;
         _workspaceManager = workspaceManager;
         _workspaceRegistrationService = registrationService;
 
         _workspaceRegistrationService.LspSolutionChanged += OnLspSolutionChanged;
         _workspaceManager.LspTextChanged += OnLspTextChanged;
     }
 
+    protected IDiagnosticSourceManager DiagnosticSourceManager { get; }
```

Comment:
this is fine.  FWIW, it's also fine to have a protected readonly field.   For internal types we're entirely inconsistent.  But my view is that properties buy is nothing as we own the entire surface area, so we dont' need the potential to swap out the prop with a different impl.

Reply:
will change to field